core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheFileInStream.java

```diff
@@ -100,12 +96,11 @@ public LocalCacheFileInStream(AlluxioURI path, OpenFilePOptions options, FileSys
   public LocalCacheFileInStream(URIStatus status, OpenFilePOptions options, FileSystem externalFs,
       CacheManager cacheManager) {
     mPageSize = externalFs.getConf().getBytes(PropertyKey.USER_CLIENT_CACHE_PAGE_SIZE);
-    mPath = new AlluxioURI(status.getPath());
     mOpenOptions = options;
     mExternalFs = externalFs;
     mCacheManager = cacheManager;
-    // Lazy init of status object
     mStatus = status;
+    mCacheQuota = status.getCacheQuota();
```

Comment:
Do we need to use the new `put` API with the cache quota?

Reply:
correct. for this part I need help from @kewang1024 to understand how to pass scope of a file here.